pkg/cmd/run/shared/shared.go

```diff
@@ -427,12 +427,16 @@ type JobsPayload struct {
 	Jobs []Job
 }
 
-func GetJobs(client *api.Client, repo ghrepo.Interface, run *Run) ([]Job, error) {
+func GetJobs(client *api.Client, repo ghrepo.Interface, run *Run, attempt uint64) ([]Job, error) {
 	if run.Jobs != nil {
 		return run.Jobs, nil
 	}
 	var result JobsPayload
-	if err := client.REST(repo.RepoHost(), "GET", run.JobsURL, nil, &result); err != nil {
+	path := fmt.Sprintf("repos/%s/actions/runs/%d/jobs", ghrepo.FullName(repo), run.ID)
+	if attempt > 0 {
+		path = fmt.Sprintf("repos/%s/actions/runs/%d/attempts/%d/jobs", ghrepo.FullName(repo), run.ID, attempt)
+	}
+	if err := client.REST(repo.RepoHost(), "GET", path, nil, &result); err != nil {
```

Comment:
While the newly constructed path should be the exact same, it would be worth continuing to use the URL provided back from the GitHub API if we are pulling the latest attempt:

```suggestion
	path := run.JobsURL

	if attempt > 0 {
		path = fmt.Sprintf("repos/%s/actions/runs/%d/attempts/%d/jobs", ghrepo.FullName(repo), run.ID, attempt)
	}
	if err := client.REST(repo.RepoHost(), "GET", path, nil, &result); err != nil {
```